src/api.ts

```diff
@@ -0,0 +1,24 @@
+import { IGroup, ISchedule } from "@/interfaces";
+
+import axios from "axios";
+
+const instance = axios.create({ // здесь мы создаем отдельное его "состояние"
+  baseURL: "https://spbgti-tools-schedule-staging.herokuapp.com/api",
+  timeout: 10000
+});
+
+export async function getGroups(): Promise<IGroup[]> {
+  return (await instance.get("/groups")).data as IGroup[];
+}
+
+export async function getScheduleById(id: string, num: string, year: string, semester: string): Promise<ISchedule[]> {
```

Comment:
I think `getScheduleById` should be named as `getSchedule` because all params are listed in the definition, and it's not the only group_id.

Also, use an explicit way to describe variable names: `id` → `groupId`, `num` → `groupNum`

Reply:
Rename for getScheduleByAPI because i have the same name of function in schedule_mod.vue

Reply:
I don't think it's a good reason for `ByAPI` renaming. Instead of using full path import, try to use module import. Something like this:

```
import 'api';

getSchedule() {
   this.schedule = api.getSchedule(...)
}
```

Reply:
In TypeScript it will be 
`import * as api from '@/api';`